src/main/ipcMain.ts

```diff
@@ -0,0 +1,52 @@
+import Electron from 'electron';
+import Logging, { Log } from '@/utils/logging';
+
+const console = Logging.background;
+
+// Intended to be passed to the replacer parameter in a JSON.stringify
+// call. Should rectify any circular references that the object you are
+// stringifying may have.
+function removeCircularReferences(property: string | symbol, value: any): any {
+  if (property === '_idlePrev') {
+    return undefined;
+  }
+
+  return value;
+}
+
+export function makeArgsPrintable(args: any[]): string[] {
+  const maxPrintableArgLength = 500;
+  const printableArgs = args.map((arg) => {
+    let printableArg = JSON.stringify(arg, removeCircularReferences);
+
+    if (printableArg.length > maxPrintableArgLength) {
+      printableArg = printableArg.slice(0, maxPrintableArgLength);
+      printableArg += '...';
+    }
+
+    return printableArg;
+  });
+
+  return printableArgs;
+}
+
+export function getIpcMainProxy(logger: Log) {
+  return new Proxy(Electron.ipcMain, {
+    get: (target, property) => {
+      if (property === 'on') {
+        return (channel: string, listener: (event: Electron.IpcMainEvent, ...args: any[]) => void) => {
+          const newListener = (event: Electron.IpcMainEvent, ...args: any[]) => {
+            const printableArgs = makeArgsPrintable(args);
+
+            logger.debug(`ipcMain: "${ channel }" triggered: ${ printableArgs.join(', ') }`);
```

Comment:
This is nitpicking, but I think it would be a little clearer if we were to change the wording here. Maybe something like `ipcMain: "example-event" triggered with arguments: arg1, arg2, arg3`.

Another separate thing here: looking at what this prints to background.log, I see a lot of `[object Object]`. For example:
```
2022-07-22T20:55:01.714Z: event settings-read in main: [object Object]
2022-07-22T20:55:02.436Z: ipcMain: "images-namespaces-read" triggered:
2022-07-22T20:55:02.445Z: ipcMain: "settings-read" triggered:
2022-07-22T20:55:02.445Z: ipcMain: "settings-read" triggered:
2022-07-22T20:55:02.445Z: event settings-read in main: [object Object]
2022-07-22T20:55:03.696Z: ipcMain: "k8s-state" triggered:
2022-07-22T20:55:03.711Z: ipcMain: "settings-read" triggered:
2022-07-22T20:55:03.711Z: ipcMain: "settings-read" triggered:
2022-07-22T20:55:03.711Z: event settings-read in main: [object Object]
```
Obviously this isn't the most helpful thing to be printing to the log files. Would it be possible to print the content of the object to the log file? Maybe a modification to `makeArgsPrintable` is needed. [This SO thread](https://stackoverflow.com/questions/1625208/print-content-of-javascript-object) may be helpful (though I wouldn't introduce indents in this case).

Reply:
Made the recommended wording changes.  The messages starting with "event settings..." are coming from background.ts which will be changed as part of a future issue.  The messages coming from ipcMain.ts seem to be displaying the arguments correctly:

> background.log:2022-08-02T16:24:01.978Z: ipcMain: "preferences-set-dirty" triggered with arguments: false
background.log:2022-08-02T16:26:03.463Z: ipcMain: "preferences-close" triggered with arguments: 
background.log:2022-08-02T16:26:05.579Z: ipcMain: "images-namespaces-read" triggered with arguments: 
images.log:2022-08-02T16:23:33.689Z: ipcMain: "do-image-pull" triggered with arguments: "busybox"
images.log:2022-08-02T16:26:22.190Z: ipcMain: "do-image-deletion" triggered with arguments: "busybox", "busybox:latest"